python/caffe/_caffe.cpp

```diff
@@ -359,10 +411,18 @@ BOOST_PYTHON_MODULE(_caffe) {
           bp::return_internal_reference<>()))
     .def("setup", &Layer<Dtype>::LayerSetUp)
     .def("reshape", &Layer<Dtype>::Reshape)
-    .add_property("type", bp::make_function(&Layer<Dtype>::type));
+    .add_property("type", bp::make_function(&Layer<Dtype>::type))
```

Comment:
Many of the members exposed in these changes don't seem specific to this PR. Could they be pulled out on their own?